pilot/pkg/kube/inject/webhook_test.go

```diff
@@ -512,6 +565,40 @@ volumes:
 	}
 }
 
+func TestReloadCert(t *testing.T) {
+	wh, cleanup := createWebhook(t, minimalSidecarTemplate)
+	defer cleanup()
+	stop := make(chan struct{})
+	defer func() { close(stop) }()
+	go wh.Run(stop)
+	checkCert(t, wh, testcerts.ServerCert, testcerts.ServerKey)
+	// Update cert/key files.
+	if err := ioutil.WriteFile(wh.certFile, rotatedCert, 0644); err != nil { // nolint: vetshadow
+		cleanup()
+		t.Fatalf("WriteFile(%v) failed: %v", wh.certFile, err)
+	}
+	if err := ioutil.WriteFile(wh.keyFile, rotatedKey, 0644); err != nil { // nolint: vetshadow
+		cleanup()
+		t.Fatalf("WriteFile(%v) failed: %v", wh.keyFile, err)
+	}
+	time.Sleep(5 * watchDebounceDelay)
```

Comment:
This is potentially flaky. Could we use https://onsi.github.io/gomega/#eventually here to (a) increase the retry multiple times with a longer duration and (b) exist early once the condition is satisfied. A few seconds upper bound seems reasonable for more consistent tests, particularly if the common case is much faster than that.